### ISSUE TITLE
Name of course to be completed shows {$a} for Student

### ISSUE THREAD

**Issue by jshrek** [ModernLMS/moodle-availability_othercompleted#33]
I am testing the plugin, and I have added course completion restriction to the top section of a course.

The Admin users sees the name of the previous course fine, but Student user sees {$a}.

See attached screenshot.
![lock](https://github.com/ModernLMS/moodle-availability_othercompleted/assets/19673842/7d99e109-7b36-498f-8512-875f8f8cfa74)

**Comment by ishara084** [comment]
I'm facing the same issue for students.

**Comment by jasimp** [comment]
i am using versio 2021033000 (2021112500) on moodle 3.11 and seeing the message as below
"Sorry, this activity is currently hidden"

**Comment by AsheniPerera** [comment]
@sam-suresh Will you able to fix this issue? We are kind of waiting for the patch.

**Comment by jshrek** [comment]
WORK AROUND: As a work around, while waiting for this bug to get fixed, I have changed the text as follows:

- Under ADMINISTRATION > LANGUAGE > LANGUAGE CUSTOMIZATION I have selected English and then Open Language Pack For Editing.

- Scroll thru the Filter Strings and look for AVAILABILITY then select **availability_othercompleted.php** then click Show Strings.

- For the **requires_complete** text, I changed it to: 
_You have completed the previous class._

This works for us because we are always preventing access until the previous class is complete.

**Comment by adevonshire78** [comment]
@sam-suresh - a fix would be greatly appreciated

**Comment by adevonshire78** [comment]
@jshrek while that is a workaround - it doesnt cater perfectly for the course or course (where a student needs to complete any of a list of courses) as this lists this "You have completed the previous class." as many times as you have optional courses to complete

**Comment by jshrek** [comment]
@adevonshire78 Yes that is why I said it "works for us". It unfortunately will not work for everybody. Hopefully bug fix will be released soon.

**Comment by jshrek** [comment]
@danielneis So I tried changing both these settings for Student role from Not Set to Allow, but it did not fix the problem. The Student already enrolled in the course and the previous course it is supposed to refer too, so this does not appear to be the issue.

**Comment by danielneis** [comment]
You may have to adjust the Authenticated User role, because the get_roles function from Moodle that is used by the plugin will test the capability on the category level, not at the course level where the user has the student role.

**Comment by jshrek** [comment]
@danielneis Just want to confirm that setting category:viewcourselist to ALLOW for the Authenticated user (not Student) did the trick and it is working now. Thank you
